### ISSUE TITLE
[1.10.2] SpongeForge on All The Mods 1 with SpongeBootstrap loads all worlds

### ISSUE THREAD

**Issue by RogueRaiden** [SpongePowered/SpongeForge#1662]
Running:
    spongeforge-1.10.2-2281-5.2.0-BETA-2464
    SpongeBootstrap-0.3.0
    forge-1.10.2-12.18.3.2316-universal
    Modpack: All The Mods v3.00
The server gets to the "preparing spawn" area and then tries to generate the spawn of every dimension, including all of the Advanced Rocketry dimensions. After a while it gets slower and slower. 
Without trying to load SpongeForge the server loads only the Overworld just fine and is completely playable.
I've tried turning off some things in the global.conf within Sponge's config folder, but doesn't seem to change anything.

**Comment by ryantheleach** [comment]
If you generate the worlds first, then add SpongeForge, does it work well then?

Also we need to know your Minecraft Forge version.

**Comment by RogueRaiden** [comment]
ive updated the post with the forge version. and the problem with that is there are like, 60 planets added with Advanced Rocketry. i could restart the server and it would go through trying to load them all again.
it never reaches a joinable state, just continues to load more dimensions. even after about 45 minutes

**Comment by axle2005** [comment]
Have you tried disabling: 
```         # Enable if this world should be loaded on startup.
        load-on-startup=true 
```

In the Sponge global config? 
(Note, Now that they have had time to load, you may need to remove the dimenison/world config for the extra worlds so it can regenerate them)

**Comment by phit** [comment]
`load-on-startup` has to be disabled in each world.conf and has no affect in the global.conf
either way this isn't a bug, see the docs for [more information](https://docs.spongepowered.org/stable/en/server/getting-started/configuration/sponge-conf.html?highlight=load-on-startup#global-properties-of-sponge)

**Comment by RogueRaiden** [comment]
alright, thanks for the information.

**Comment by arlojeremy** [comment]
This is not resolved. I have the same issue in 1.11.2, and it seems to revolve around the fact that Advanced Rocketry has so many dimensions and it tries to load all the spawns, as OP said. I had Nucleus and Nucleus Mixins installed, but even when they are disabled I have the same issue.

_Sidebar: Main reason I am using Sponge + Nucleus is to pregenerate chunks for smoother gameplay._

I checked every `world.conf` created by sponge forge for all 37 dimensions created by Advanced Rocketry, and every `world.conf` file in each `DIM###` directory has `load-on-startup` set to false.

It still tries to generate the spawn for each dimension in Advanced Rocketry, eventually crashing when the memory limit is reached, before the server fully initializes.

I tried also setting `load-on-startup` to false in the `dimension.conf` under `worlds/advancedrocketry/planet`, and in fact EVERY conf I can find related to sponge and Advanced Rocketry.

It STILL tries to generate all spawns on server load. Am I missing something here?

Thanks.

**Comment by Zidane** [comment]
@phit @EZKatnipz @arlojeremy 

Completely spaced when I read what @phit said. Also try:

```generate-spawn-on-load```

Set that to false

**Comment by RogueRaiden** [comment]
from my own testing, it seems that if the dimension hasnt been loaded yet, it will try to load it. once it has done so, you can then set that dimension's configs to false, and it shouldn't load anymore. I had done all this in a test server before trying to load Sponge into my main server, and moved the configs over with all the changes but it decided to change the configs back for most dimensions and generate them. I then had to go back and change all the configs back to false, and they no longer generate on startup.
Only a dimensions world.conf will change it, the dimension.conf won't (from my experience)

It would be nice for the main config to be a default for newly generated ones to use so that users dont have to wait 2 hours for their server to generate 70 dimensions with Advanced Rocketry in order to disable their starups.

**Comment by arlojeremy** [comment]
edit: When I put 'generate-spawn-load' in the following comment, I really meant 'generate-spawn-on-load', which is what I did the replacement with, I only mistyped it in this comment.

Hi @Zidane. I went the sponge config folder, and using Notepad++ file search and replace replaced all `generate-spawn-load=true` to `generate-spawn-load=false` and tried restarting the server (after checking a few to make sure the replace worked). I also changed that setting in any other config I could find.

Unfortunately, it still started generating every spawn dimension again, and when I went back, it was changing all of the individual dimensions' `world.conf` file to read `generate-spawn-load=true`, but it did _NOT_ change `load-on-startup` from false to true. Sounds similar to what @EZKatnipz  was saying?

Happy to provide more information or testing for confirmation on this with video of what I'm doing, although it'll have to wait until after the weekend as I'm working two jobs. Thanks.

**Comment by Zidane** [comment]
@arlojeremy 

It's ```generate-spawn-on-load```

**Comment by arlojeremy** [comment]
@Zidane Whoops, my bad. I actually only mistyped it in the comment above, not when I was doing the actual search and replace.

Sorry, I haven't had time to work on this much, but what I don't get is that I set up a server (this time in 1.10.2) that ONLY had Advanced Rocketry, SpongeForge and Nucleus+Mixins and it started up fine without going nuts pre-generating all the AR spawns. Started a new 1.10.2 heavily modded server and added the same version of SpongeForge and Nucleus+Mixins and it goes and tries to pre-gen everything all over again.

Modding is hard lol. Thanks.

**Comment by arlojeremy** [comment]
Quick update.

On that minimal server installation I swapped in and out a few different mods with a minimal spongeforge setup. With the latest `spongeforge`, `nucleus`, `nucleus-mixins` and `advanced rocketry`, it appears that adding in Astral Sorcery was the cause of the generating issue. After adding it, the issue replicated.

To make sure it wasn't just an `astral sorcery` and `advanced rocketry` conflict, I ran a server with only those two mods, and it worked fine.

So, it appears to be a problem when running those two mods along with `spongeforge`, but I could not find any settings for `astral sorcery` or `advanced rocketry` that might affect this problem.

Thanks.
